script/lint.js

```diff
@@ -99,20 +100,26 @@ const LINTERS = [{
   test: filename => filename.endsWith('.js') || filename.endsWith('.ts'),
   run: async (opts, filenames) => {
     const eslint = new ESLint({
-      cache: true,
+      // Do not use the lint cache on CI builds
+      cache: !process.env.CI,
+      cacheLocation: `node_modules/.eslintcache.${crypto.createHash('md5').update(fs.readFileSync(__filename)).digest('hex')}`,
```

Comment:
wait, what? why do we need to base the cache _location_ on the hash of the filename?

why is this in `node_modules/`?

Reply:
It's in `node_modules` so that it doesn't pollute the top level directory, this is just a safety measure.  There have been times where I've updated this script and the way it calls eslint and it doesn't bust the cache.  Seems like a safe idea.

Reply:
Could the cache be put under `.eslintcache` rather than `node_modules/`? Keeps it more visible and if it stops working as intended there's a better chance it would get caught since it would fall back to `.eslintcache` and if that's a directory, would probably choke on that.

`cacheLocation: `.eslintcache/${crypto.createHash('md5').update(fs.readFileSync(__filename)).digest('hex')}`,` seems to work as expected and makes the `.eslintcache` directory even if it doesn't exist. And everything is still covered by the `.gitignore` pattern.

Reply:
I think `node_modules` it the correct place for these, it's where things like typescripts and webpacks cache go as well. It's out of the way and no user should ever need to look at it 🤷